### PR TITLE
Make `FLOAT` SQL type map to `Float32` rather than `Float64`

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

minor fix and followup for https://github.com/apache/arrow-datafusion/pull/1390

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/src/sql/planner.rs [hunk 1]
```diff
@@ -1989,10 +1989,10 @@ fn extract_possible_join_keys(
...
     match sql {
         SQLDataType::Boolean => Ok(DataType::Boolean),
-        SQLDataType::SmallInt(_display) => Ok(DataType::Int16),
-        SQLDataType::Int(_display) => Ok(DataType::Int32),
-        SQLDataType::BigInt(_display) => Ok(DataType::Int64),
-        SQLDataType::Float(_) => Ok(DataType::Float64),
+        SQLDataType::SmallInt(_) => Ok(DataType::Int16),
+        SQLDataType::Int(_) => Ok(DataType::Int32),
+        SQLDataType::BigInt(_) => Ok(DataType::Int64),
+        SQLDataType::Float(_) => Ok(DataType::Float32),
```
Comment (liukun4515) [inline datafusion/src/sql/planner.rs:1995]: The SQL FLOAT datatype is corresponding to the Arrow Float32

Comment (alamb) [reply datafusion/src/sql/planner.rs:1995]: This PR changes `FLOAT` to use `Float32` (rather than `Float64`). Was that the intention?

Comment (liukun4515) [reply datafusion/src/sql/planner.rs:1995]: Yes.
In the datafusion [README.md](https://github.com/apache/arrow-datafusion/blob/6955a8d9279bf06311fe0d9d7428854b0601059d/README.md?plain=1#L345), the SQL float is corresponding to the arrow-FLOAT32.

**Comment by liukun4515** [comment]
Yes.
In the datafusion [README.md](https://github.com/apache/arrow-datafusion/blob/6955a8d9279bf06311fe0d9d7428854b0601059d/README.md?plain=1#L345), the SQL float is corresponding to the arrow-FLOAT32.

@alamb

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @liukun4515

**Pull request merged**